src/backend/utils/misc/queryenvironment.c

```diff
@@ -261,7 +264,14 @@ bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List
 	Datum v1 = 0, v2 = 0, v3 = 0, v4 = 0;
 
 	if (sql_dialect != SQL_DIALECT_TSQL)
-		return false;
+	{
+		/*
+		 * It is possible we are inside a non tsql proc/function
+		 * where sql_dialect is temporarily set to PG.
+		 */
+		if (!in_non_tsql_proc_hook || !in_non_tsql_proc_hook())
```

Comment:
what do we need this change? If dialect is not TSQL, this hook indicates that there is no PG function active in call stack, how does it influence ENR scan?

Reply:
Hi Surendra, good question.

There are cases when we overwrite the tsql_dialect. In the case for identity, we call nextval_internal from `fmgr_security_definer` where tsql_dialect gets overriden temporarily. When this happens, we execute `nextval_internal` in non tsql dialect.

Reply:
This is not ideal. We are adding objects in ENR which relates to temp table/variables only. Even we skip adding temp table with UDT to ENR to avoid UDT dependency. Given sequence is internal to temp object only. Is it possible to avoid this dependency on PG function call and rather call the function (nextval_internal) directly?